### ISSUE TITLE
Support for building .NET Core 2.1.4 SDK

### ISSUE THREAD

**Issue by tmds** [dotnet/source-build#258]
I don't know if there is already some support to build this.

@dotnet/source-build-contrib 

- [ ] Port dev/release/2.0 to master 
- [ ] repo owners port their patch removal from 2.0 servicing to their master (2.1 branch).
- [ ] update the master SHA to masters of submodule repos

**Comment by weshaggard** [comment]
We have started working on this by getting all our changes into the master branches. Once I get all those changes into our core repo's I'm going to update source-build master branch to pull in those master/2.1 branches.

**Comment by omajid** [comment]
AFAIK @tmds filed this to track staying up to date with https://github.com/dotnet/designs/pull/18. But it looks like https://github.com/dotnet/source-build/issues/259 is already tracking that. Is there anything special we need to track here? @tmds can we close this?

**Comment by tmds** [comment]
Looks like @dleeapho and myself created an issue about this at about the same time.
Perhaps we can track our ability to build 2.1 here? WDYT @omajid ?

**Comment by dleeapho** [comment]
I decided to close #259 and track the work here.

**Comment by omajid** [comment]
@tmds I was going to say that since Dan and his team are doing the work, they should get to choose which bug they want to use to track this :)

Looks like @dleeapho was kind enough to switch to this one. Thanks, @dleeapho !

**Comment by weshaggard** [comment]
https://github.com/dotnet/source-build/pull/268 is the first step of moving all the core repo's to master pointers for supporting a 2.1 build.

**Comment by omajid** [comment]
Hey folks, any updates on having a source-build release for the 2.1 sdk?

**Comment by chcosta** [comment]
@crummel

**Comment by dleeapho** [comment]
@omajid once the submodule repos branch for 2.1 we'll be updating source-build.

**Comment by omajid** [comment]
@dleeapho Hey. Sorry if it was not clear, but I was asking about a release that corresponds to the 2.1 *SDK* (not runtime). I see 2.1.2 as released already [here](https://www.microsoft.com/net/download/linux), so I suppose the repositories must have branched already?

**Comment by tmds** [comment]
I have updated the title to better indicate this issue tracks support for building the 2.1.4 SDK. That is: the SDK that uses .NET Core Runtime 2.0 and adds C# 7.2 support.

I think when we version this in Linux packages, we want to use the new versioning scheme already (https://github.com/dotnet/designs/pull/29). So I think it makes sense to version this 2.0.104 (see https://github.com/dotnet/designs/pull/29#issuecomment-361169564).

I created a separate issue to track support for the new 2.1 runtime and its associated SDK: https://github.com/dotnet/source-build/issues/315.
